include/zephyr/senss/senss_datatypes.h

```diff
@@ -0,0 +1,313 @@
+ * more than 4295 seconds, the sensor subsystem core will split them
+ * across multiple instances of the readings structure, and send multiple
+ * events.
+ *
+ * This concept is borrowed from CHRE:
+ * https://cs.android.com/android/platform/superproject/+/master:\
+ * system/chre/chre_api/include/chre_api/chre/sensor_types.h
+ */
+struct senss_sensor_value_header {
+	/** base timestamp of this data readings, unit is micro seconds */
+	uint64_t base_timestamp;
+	/** count of this data readings */
+	uint16_t reading_count;
+};
+
+/**
+ * @brief Sensor value data structure types based on common data types.
+ * Suitable for common sensors, such as IMU, Light sensors and orientation sensors.
+ *
+ * Sensor subsystem will use scaled fixed point data structure for all sensor values,
+ * aligned the HID spec, using the format v*10^x to present the decimal value,
+ * where the v is integer number, either int8/uint8, int16/uint6, or int32/uint32, depends on
+ * required sensor data precision.
+ *
+ * The scale unit exponent x is int8 type int8 type with encoding meanings (page 68 of
+ * HID spec https://usb.org/sites/default/files/hutrr39b_0.pdf):
+ *
+ * Unit Exponet argument       Power of Ten (Scientific Notation)
+ * 0x00			       1 * 10E0
+ * 0x01			       1 * 10E1
+ * 0x02			       1 * 10E2
+ * 0x03			       1 * 10E3
+ * 0x04			       1 * 10E4
+ * 0x05			       1 * 10E5
+ * 0x06			       1 * 10E6
+ * 0x07			       1 * 10E7
+ * 0x08			       1 * 10E-8
+ * 0x09			       1 * 10E-7
+ * 0x0A			       1 * 10E-6
+ * 0x0B			       1 * 10E-5
+ * 0x0C			       1 * 10E-4
+ * 0x0D			       1 * 10E-3
+ * 0x0E			       1 * 10E-2
+ * 0x0F			       1 * 10E-1
+ *
+ * So, we can have below data present ranges:
+ *
+ * int8 type v:    [-128, 127] *10^[-8, 7]
+ * uint8 type v:   [0,	255] * 10^[-8, 7]
+ * int16 type v:   [-32768, 32767] * 10^[-8, 7]
+ * uint16 type v:  [0,	65535\r] * 10^[-8, 7]
+ * int32 type v:   [-2147483648,  2147483647] * 10^[-8, 7]
+ * uint32 type v:  [0,	4294967295] * 10^[-8, 7]
+ *
+ * To simple the data structure definition and save store memory, only keep v in code definitions,
+ * scale exponent x will defined in doc and spec,  but not explicitly present in code, for scenarios
+ * which need transfer to decimal value, such as in a algorithm process, need base on the sensor
+ * type and according the doc/spec to get the right scale exponent value x.
+ *
+ * A example in doc and spec can be like:
+ * 3D accelerometer:
+ *	Data Fields    Type	Unit	    Unit Exponent   Typical Range     Description
+ *	data[0]        int32	micro g     -6		    +/-4*10^6	      x axis acceleration
+ *	data[1]        int32	micro g     -6		    +/-4*10^6	      y axis acceleration
+ *	data[2]        int32	micro g     -6		    +/-4*10^6	      z axis acceleration
+ *
+ * Ambient Light:
+ *	Data Fields    Type	Unit	    Unit Exponent   Typical Range     Description
+ *	data[0]        uint32	milli lux   -3		    [0, 10000] *10^3  Ambient light lux level
```

Comment:
The unit exponent is likely to change in something like a light sensor that covers less than 1 lux indoors, but may be tens of thousands of lux outdoors, especially with 16 bit values.

I think this information is important to convey **in code** inside the sensor driver itself, even if it isn't part of every sensor data sample, and we should be able to query any sensor driver for basic config information like the exponent and base type. Hard-coding exponent assumptions (that are only available in documentation)  into sensor data consumer feels like a limitation we can easily avoid.

I haven't gone through this PR exhaustively yet (commenting now as I work through it), but what comes to mind for me is:

1. The need to get a 'config' struct for this sensor on request that includes the unit exponent, amongst other things
2. Have a callback we can optionally enable whenever the sensor driver changes something with the device config internally, such as the gain on an accelerometer or light sensor being changed by a SW auto-gain algorithm, which will trigger the callback after the register changes.

Reply:
Since we have uint32 for lux, so fixed unit exponent still can cover the indoor/outdoor situations.
But I think your idea of dynamic exponent unit is valuable to think about. 

Just like you mentioned, I used think about to design a API for sensors to describe their data fields, such as, type, size, unit, exponent unit, ranges and so on. But found its not easy, and will make things complexity and trivializes , since sensor data are very different between different sensor types. 

So, to make things more simple, and make sensor types more stable and standardized, I think a spec is more better, since code is easy changing. :)